app/src/main/java/com/trunghoang/aroundhere/ui/discover/DiscoverFragment.java

```diff
@@ -79,12 +95,66 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         }
     }
 
+    @Override
+    public void onResume() {
+        super.onResume();
+    }
+
     @Override
     public void onDetach() {
         super.onDetach();
         mListener = null;
     }
 
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
+        if (requestCode != PERMISSION_REQUEST_FINE_LOCATION) return;
+        if (grantResults.length == HAS_ONE_PERMISSION
+                && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            mPresenter.start();
+        } else {
+            //TODO: Show request failed status
+        }
+    }
+
+    @Override
+    public void setPresenter(@NonNull DiscoverContract.Presenter presenter) {
+        mPresenter = presenter;
+    }
+
+    @Override
+    public void showPlaces(List<Place> places) {
+        mPlacesAdapter.setPlaces(places);
+        showSearchResultCount(places.size());
+    }
+
+    @Override
+    public void showLoadingPlacesError() {
+    }
+
+    @Override
+    public boolean isActive() {
+        return isAdded();
+    }
+
+    @Override
+    public boolean isLocationPermissionGranted() {
+        return (ActivityCompat.checkSelfPermission(mContext,
+                Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED);
+    }
+
+    @Override
+    public void requestLocationPermission() {
+        requestPermissions(new String[]{Manifest.permission.ACCESS_FINE_LOCATION},
```

Comment:
define new String[]{Manifest.permission.ACCESS_FINE_LOCATION} is final static variable